lib/puppet/util/windows/adsi.rb

```diff
@@ -120,20 +120,51 @@ def self.parse_name(name)
       return account, domain
     end
 
-    def native_user
-      @native_user ||= Puppet::Util::Windows::ADSI.connect(self.class.uri(*self.class.parse_name(@name)))
+    def get_sids(adsi_child_collection)
+      sids = []
+      adsi_child_collection.each do |m|
+        sids << Puppet::Util::Windows::SID.octet_string_to_sid_object(m.objectSID)
+      end
+
+      sids
     end
 
-    def sid
-      @sid ||= Puppet::Util::Windows::SID.octet_string_to_sid_object(native_user.objectSID)
+    def name_sid_hash(names)
+      return {} if names.nil? or names.empty?
```

Comment:
Please let's not use `or` flow control here, instead `return {} if names.nil? || names.empty?`